test/Sema/result_depends_on_errors.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift  -enable-builtin-module
```

Comment:
Hopefully this test now fails without `-enable-experimental-feature NonEscapableTypes`.
Then you'll need `REQUIRES: asserts`